demos/bounded_heat/bounded_heat.py.rst

```diff
@@ -0,0 +1,202 @@
+Solving the Heat Equation with Bounds Constraints
+=================================================
+
+In this demo we solve the simple heat equation with bounds constraints applied uniformly in time and space. 
+Consider the simple heat equation on :math:`\Omega = [0,1]\times [0,1]`, with boundary :math:`\Gamma`:
+
+.. math::
+
+    u_t - \Delta u &= f
+
+    u & = g \quad \textrm{on}\ \Gamma
+
+for some known functions :math:`f` and :math:`g`. The solution will be some function :math:`u\in V`, for 
+a suitable function space :math:`V`.
+
+The weak form is found by multiplying by an arbitrary test function :math:`v\in V` and integrating over :math:`\Omega`. 
+We then have the variational problem of finding :math:`u:[0,T]\rightarrow V` such that 
+.. math::
+
+    (u_t, v) + (\nabla u, \nabla v) = (f, v)
+
+This demo uses particular choices of the functions :math:`f` and :math:`g` to be defined below.
+
+First, we must import firedrake and certain items from Irksome: ::
+
+    from firedrake import *
+    from irksome import (Dt, MeshConstant, RadauIIA, TimeStepper)
+
+We also need some UFL tools in order to manufacture a solution: ::
+
+    from ufl.algorithms import expand_derivatives
+
+Finally, numpy provides us with the upper bound of infinity: ::
+
+    import numpy as np
+
+We first define the mesh and the necessary function space. We choose 
```

Comment:
Expand the comment on Bernstein: constraining the degrees of freedom for the Bernstein basis guarantees positivity of the solution, but this is not true for Lagrange.  The nodal values there can be constrained, but this doesn't guarantee uniform positivity.